sql/table.cc

```diff
@@ -6029,7 +6029,7 @@ bool TABLE::add_tmp_key(Field_map *key_parts, bool invisible,
     s->max_key_length = std::max(s->max_key_length, key_len);
     s->key_parts += key_part_count;
     assert(s->keys < s->max_tmp_keys);
-    sprintf(s->key_names[s->keys].name, "<auto_key%d>", s->keys);
+    sprintf(s->key_names[s->keys].name, "<auto_key%d>", s->temp_table_key_id++);
```

Comment:
[optional]: if you want to reduce testcase diff, may be we can use ++s->temp_table_key_id.

Reply:
Most probably it would increase the number of tests to be re-recorded because the 1st auto_key will be <auto_key1> always. I see a lot of tests results have <auto_key0> inside

Reply:
Hmm, this is weird. Sometimes keys with 0 and sometimes key with 1. Ignore it.

Reply:
my bad, because of dup key issue, it might be re-using either 0 or 1.

Reply:
Note that there may be other keys (not <auto_keyN>) already there (like <auto_distinct_key>), so the 1st auto_keyN create had 1 in the name before the fix (this is why re-record was needed - now it has 0 in its name). 
But there can be also a situation when there are no keys, so 1st auto_key would have 0 in name before the change. If we changed it as you proposed, it would have 1 in the name, and it won't be aligned with already recorded test results (other tests)

Reply:
Thanks for explaining the reason for the difference of key IDs 0 and 1. Please leave it as is.